File: test_script.js
const axios = require('axios');

async function test() {
  try {
    const loginResp = await axios.post('http://localhost:8000/api/usuarios/login', {
      email: 'smoke_admin@example.com',
      'contraseña': 'adminpassword'
    });
    const token = loginResp.data.access_token;
    const authHeaders = { headers: { Authorization: `Bearer ${token}` } };
    
    const initialItems = await axios.get('http://localhost:8000/api/items', authHeaders);
    const ic = initialItems.data.length;

    const pResp = await axios.post('http://localhost:8000/api/pilotos', { nombre: 'P' }, authHeaders);
    const pid = pResp.data.id;

    const mResp = await axios.post('http://localhost:8000/api/motocicletas', {
      modelo: 'M',
      piloto_id: pid,
      trabajos_reparacion: ['Cambio de bomba', 'Ajuste electrico'],
      servicio_ids: []
    }, authHeaders);
    const mid = mResp.data.id;

    const finalItems = await axios.get('http://localhost:8000/api/items', authHeaders);
    const fc = finalItems.data.length;

    const eResp = await axios.get('http://localhost:8000/api/estados', authHeaders);
    const names = eResp.data
      .filter(e => e.motocicleta_id === mid)
      .map(e => e.nombre);

    console.log(`IC: ${ic}, FC: ${fc}, Names: ${names.join(', ')}`);
    if (ic === fc && names.includes('Cambio de bomba') && names.includes('Ajuste electrico')) {
      console.log('PASS');
    } else {
      console.log('FAIL');
    }
  } catch (err) {
    if (err.response && err.response.status === 401 && err.config.url.endsWith('/login')) {
       // Try without the special char just in case
        try {
            const loginResp = await axios.post('http://localhost:8000/api/usuarios/login', {
                email: 'smoke_admin@example.com',
                'password': 'adminpassword'
            });
            // recursive simple call if needed or just handle here. Let's just log and fail for now.
        } catch (e2) {
             console.error(err.response ? err.response.data : err.message);
             process.exit(1);
        }
    }
    console.error(err.response ? err.response.data : err.message);
    process.exit(1);
  }
}

test();

File: test_script.py
import requests

url = "http://localhost:8000/api/usuarios/login"
payload = {
    "email": "smoke_admin@example.com",
    "contraseña": "adminpassword"
}
response = requests.post(url, json=payload)
if response.status_code != 200:
    print(f"Login failed: {response.text}")
    exit(1)

token = response.json()["access_token"]
headers = {"Authorization": f"Bearer {token}"}

# Get initial item count
ic = len(requests.get("http://localhost:8000/api/items", headers=headers).json())

# Create piloto
p_resp = requests.post("http://localhost:8000/api/pilotos", headers=headers, json={"nombre": "P"})
pid = p_resp.json()["id"]

# Create motocicleta
m_payload = {
    "modelo": "M",
    "piloto_id": pid,
    "trabajos_reparacion": ["Cambio de bomba", "Ajuste electrico"],
    "servicio_ids": []
}
m_resp = requests.post("http://localhost:8000/api/motocicletas", headers=headers, json=m_payload)
mid = m_resp.json()["id"]

# Get final item count
fc = len(requests.get("http://localhost:8000/api/items", headers=headers).json())

# Get estados
e_resp = requests.get("http://localhost:8000/api/estados", headers=headers)
estados = e_resp.json()
names = [e["nombre"] for e in estados if e["motocicleta_id"] == mid]

print(f"IC: {ic}, FC: {fc}, Names: {names}")
if ic == fc and "Cambio de bomba" in names and "Ajuste electrico" in names:
    print("PASS")
else:
    print("FAIL")
